### ISSUE TITLE
Move library to new location in GROMACS tree

### ISSUE THREAD

**Issue by giacomofiorin** [Colvars/colvars#460]
The current location in the [patched releases](https://github.com/Colvars/gromacs) is `src/gromacs/colvars`, which is really not ideal. While it makes it a bit simpler to build the release, the code organization should clearly reflect where the components are developed. A more suitable location would be `src/external/colvars`.

I put a tentative change for the 2021 patch in [this branch](https://github.com/Colvars/gromacs/tree/release-2021-colvars-in-external). I can re-generate the corresponding patches, but @HubLot @jhenin it's best to get your input first, particularly considering the MDModules version of the interface, which I didn't test.

The core of it is the patch below, which allows including headers and linking the library without need to put it under the `gromacs` folder.
```
diff --git a/src/gromacs/CMakeLists.txt b/src/gromacs/CMakeLists.txt
index a4430e9dd6..81a764b554 100644
--- a/src/gromacs/CMakeLists.txt
+++ b/src/gromacs/CMakeLists.txt
@@ -142,6 +142,10 @@ if (WIN32)
 endif()
 list(APPEND libgromacs_object_library_dependencies thread_mpi)
 
+file(GLOB COLVARS_SOURCES ${PROJECT_SOURCE_DIR}/src/external/colvars/*.cpp)
+add_library(colvars OBJECT ${COLVARS_SOURCES})
+list(APPEND libgromacs_object_library_dependencies colvars)
+
 configure_file(version.h.cmakein version.h)
 if(GMX_INSTALL_LEGACY_API)
   install(FILES
@@ -189,6 +193,8 @@ else()
     add_library(libgromacs ${LIBGROMACS_SOURCES})
 endif()
 
+target_include_directories(libgromacs PRIVATE ${PROJECT_SOURCE_DIR}/src/external/colvars)
+
 # Add these contents first because linking their tests can take a lot
 # of time, so we want lots of parallel work still available after
 # linking starts.
```

**Comment by HubLot** [comment]
Hi Giacomo,

The changes seems correct to me. We will need to modify the `update_colvars_code.sh` too
2 points though:
  - The  2 includes of colvars_proxy in `runner.cpp` and `sim_util.cpp` should be placed before the gromacs include as it's done elsewhere for similar external codes.
  - At some point, we may need to create a special cmake file under their `cmake` folder for colvars and lepton. We should discuss with them the proper way to do this. Anyway it's beyond the scope of this issue imo.

**Comment by giacomofiorin** [comment]
@HubLot Nope, we need the colvarproxy header included *after* the GROMACS includes, so that the GROMACS data types are defined.

If you want to be able to define the `colvarproxy_gromacs` class before the GROMACS types, I think the only reasonable solution would be using forward declarations and pointers as the members instead of the objects themselves.

**Comment by giacomofiorin** [comment]
Closed by #465
